ibis/util.py

```diff
@@ -522,6 +522,21 @@ def gen_name(namespace: str) -> str:
     return f"ibis_{namespace}_{uid}"
 
 
+def gen_name_from_path(path: str | Path, file_type: str) -> str:
+    """Create a user-friendly unique identifier from a file path.
+
+    Examples
+    --------
+    >>> gen_name_from_path("path/to/myfile.csv", "csv")  # doctest: +ELLIPSIS
+    'ibis_read_csv_myfile_...'
+    """
+    root, _ext = os.path.splitext(path)
+    basename = os.path.basename(root)
+    basename = re.sub(r"[^a-zA-Z0-9_]", "_", basename).strip("_")
```

Comment:
```suggestion
    basename = re.sub(r"\W", "_", basename).strip("_")
```

the compilation cost for your regex is pretty low anyway, but this one is prebuilt

Reply:
I think I would slightly prefer to keep this as is, this code path is gonna get hit at most ~10k?? times in a session, so I think we don't need to worry about perf, and at least personally, I have to go look up what \W is, but the current spelling requires no lookup. If you feel strongly I will change it though <3

<img width="795" alt="image" src="https://github.com/user-attachments/assets/c855d51e-cfda-46a4-bf5a-7ea6b7f50b75">

Reply:
I dunno, those 4 nanoseconds look enticing!

Yeah, we can keep this as is 😅